### ISSUE TITLE
Export variable from Resource script lost when duplicating resource in exported build

### ISSUE THREAD

**Issue by dbsGen** [godotengine/godot#3417]
This issue happened under the situation of the script is binding on a Resource(or subtype of Resource) and export the project with script compiled.

**Comment by reduz** [comment]
please make a small example project showing the problem, otherwise it's difficult to figure out from what you describe

**Comment by dbsGen** [comment]
@reduz I show you this problem.
[script_problem.zip](https://github.com/godotengine/godot/files/113487/script_problem.zip)

Run it in the editor it say "Has script".
![19b789f4-08ae-4225-8413-a688d336a416](https://cloud.githubusercontent.com/assets/1573606/12741761/4f4d4666-c9ba-11e5-8a5d-da4f413e8b41.png)
Then, export it with script compiled.
![e0e08f0d-1f06-4268-b4d3-521e0be09ac4](https://cloud.githubusercontent.com/assets/1573606/12741770/69c34bb2-c9ba-11e5-94fd-e3eab60456b9.png)
And run the exported application, it say "No script".
![d85a526a-0a59-4ce0-b0ce-81cf937637c2](https://cloud.githubusercontent.com/assets/1573606/12741782/7aa0afe2-c9ba-11e5-9a38-2f45be01adab.png)

**Comment by Chaosus** [comment]
Still exist in 3.1
[HasScript3.zip](https://github.com/godotengine/godot/files/2393698/HasScript3.zip)

**Comment by akien-mga** [comment]
Fixed the title, the issue is a bit more specific than just losing a bound script. What happens is that the export variable is lost when duplicating the resource in the exported build only.

Here's an updated project that demonstrates it (valid in current master): 
[HasScript3.zip](https://github.com/godotengine/godot/files/2681076/HasScript3.zip)

In editor:
```
res_data script: [GDScript:1062]
res_data has script_data: True
res_dup script: [GDScript:1071]
res_dup has script_data: True
```

Once exported:
```
res_data script: [GDScript:1029]
res_data has script_data: True
res_dup script: [GDScript:1038]
res_dup has script_data: Null
```

**Comment by akien-mga** [comment]
Still reproducible in current master branch (966c68ba).

**Comment by KoBeWi** [comment]
Still valid in 3.2.3

**Comment by jkb0o** [comment]
I guess this should be fixed by #41925
